identity-service/src/routes/tiktok.js

```diff
@@ -57,9 +64,159 @@ module.exports = function (app) {
       urlAccessToken += '&code=' + code
       urlAccessToken += '&grant_type=authorization_code'
 
-      const resp = await axios.post(urlAccessToken)
+      try {
+        // Fetch user's accessToken
+        const accessTokenResponse = await axios.post(urlAccessToken)
+        const { access_token: accessToken } = accessTokenResponse.data
+
+        // Fetch TikTok user from the TikTok API
+        const userResponse = await axios.post(
+          `https://open-api.tiktok.com/user/info/?access_token=${accessToken}`,
+          {
+            fields: [
+              'open_id',
+              'display_name',
+              'profile_deep_link',
+              'is_verified'
+            ]
+          }
+        )
+
+        const { data, error } = userResponse.data
+
+        if (error.code) {
+          return errorResponseBadRequest(error.message)
+        }
+
+        const { user: tikTokUser } = data
+
+        const existingTikTokUser = await models.TikTokUser.findOne({
+          where: {
+            uuid: tikTokUser.open_id,
+            blockchainUserId: {
```

Comment:
This is interesting, what does this check?

Reply:
This checks that the TikTok user exists and is not associated with a User. But it's probably not necessary anymore because the associate route already makes this assertion